<a href="https://colab.research.google.com/github/JaimeTorio/UFV_Visualizacion/blob/main/Ejercicios%20de%20clase/%20Clase2b.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# **Clase 18/10/2024**

In [1]:
import pandas as pd
df = pd.read_csv('datos_ejercicio_ventas.csv')
print(df.head())


         COUNTRY        SUBBRAND  YEAR  MONTH     SCENARIO FORECAST  \
0       Portugal     Lipton (L3)  2023     12  AI_forecast  AI_P02F   
1  Great Britain     Lipton (L3)  2023     12  AI_forecast  AI_P10F   
2          Spain  Pepsi Max (L3)  2023     12  AI_forecast  AI_P09F   
3  Great Britain        7up (L3)  2024     12  AI_forecast  AI_P10F   
4        Hungary     Lipton (L3)  2023      9  AI_forecast  AI_P03F   

   FORECAST_YEAR         AMOUNT  
0         2023.0  754356.237194  
1         2023.0  560030.558029  
2         2023.0   88501.980847  
3         2023.0  363224.511516  
4         2023.0  396176.120491  


In [6]:
# 1. Tamaño del Dataset
dataset_size = df.shape
print(f'Tamaño del dataset: {dataset_size}')

Tamaño del dataset: (18666, 8)


In [7]:
# 2. Número de Actuals y Forecast
actuals_forecast_count = df['SCENARIO'].value_counts()
print(f'Número de Actuals y Forecast: \n{actuals_forecast_count}')

Número de Actuals y Forecast: 
SCENARIO
AI_forecast    17766
actual           900
Name: count, dtype: int64


In [18]:
# Asegurarse de que los valores de 'FORECAST_YEAR', 'YEAR', 'FORECAST_MONTH', y 'MONTH' están presentes
df['PREDICTED_DATE'] = df['FORECAST_YEAR'] + (df['FORECAST_MONTH'] - 1) / 12
df['ACTUAL_DATE'] = df['YEAR'] + (df['MONTH'] - 1) / 12

# Calcular la diferencia (HORIZON) entre la fecha predicha y la fecha actual
df['HORIZON'] = df['ACTUAL_DATE'] - df['PREDICTED_DATE']

# Convertir el horizonte a meses
df['HORIZON_MONTHS'] = df['HORIZON'] * 12

# Filtrar solo los registros con predicciones (AI_forecast)
horizonte_forecast = df[df['SCENARIO'] == 'AI_forecast'][['COUNTRY', 'SUBBRAND', 'YEAR', 'MONTH', 'HORIZON_MONTHS']]

# Mostrar los resultados de los horizontes en meses
horizonte_forecast.head()

COUNTRY        SUBBRAND  YEAR  MONTH  HORIZON_MONTHS
0       Portugal     Lipton (L3)  2023     12            10.0
1  Great Britain     Lipton (L3)  2023     12             2.0
2          Spain  Pepsi Max (L3)  2023     12             3.0
3  Great Britain        7up (L3)  2024     12            14.0
4        Hungary     Lipton (L3)  2023      9             6.0

In [19]:
# 4. Número de países y productos
num_countries = df['COUNTRY'].nunique()
num_products = df['SUBBRAND'].nunique()
print(f'Número de países: {num_countries}')
print(f'Número de productos: {num_products}')

Número de países: 9
Número de productos: 6


In [20]:
# 5. Histórico de datos (Actuals y Forecast)
historical_data_range = df[['YEAR', 'MONTH']].drop_duplicates().sort_values(by=['YEAR', 'MONTH'])
print('Histórico de datos:')
print(historical_data_range.head())

Histórico de datos:
     YEAR  MONTH
34   2023      1
397  2023      2
118  2023      3
9    2023      4
57   2023      5


In [21]:
# 6. Forecasts distintos
distinct_forecasts = df['FORECAST'].nunique()
print(f'Forecasts distintos: {distinct_forecasts}')

Forecasts distintos: 12


1. **Tamaño del Dataset**: 18,666 filas y 8 columnas.
2. **Número de Actuals y Forecast**:
   - AI_forecast: 17,766 registros.
   - Actuals: 900 registros.
3.
- Portugal (Lipton): Predicción hecha 10 meses antes de diciembre de 2023.

- Great Britain (Lipton): Predicción hecha 2 meses antes de diciembre de 2023.

- España (Pepsi Max): Predicción hecha 3 meses antes de diciembre de 2023.

- Great Britain (7up): Predicción hecha 14 meses antes de diciembre de 2024.

- Hungary (Lipton): Predicción hecha 6 meses antes de septiembre de 2023.
4. **Número de países**: 9 países.
5. **Número de productos**: 6 productos (submarcas).
6. **Histórico de datos (Actuals y Forecast)**: El rango de datos históricos comienza en enero de 2023.
7. **Forecasts distintos**: Hay 12 tipos de forecast distintos.
